app/init/websocket_manager.ts

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {AppState, AppStateStatus} from 'react-native';
+
+import type {ServerCredential} from '@typings/credentials';
+import WebSocketClient from '@app/client/websocket';
+import {handleClose, handleEvent, handleFirstConnect, handleReconnect} from '@actions/websocket';
+import NetInfo, {NetInfoState} from '@react-native-community/netinfo';
+import {setCurrentUserStatusOffline} from '@actions/local/user';
+import {queryWebSocketLastDisconnected} from '@app/queries/servers/system';
+import DatabaseManager from '@database/manager';
+
+class WebsocketManager {
+    private clients: Record<string, WebSocketClient> = {};
+    private previousAppState: AppStateStatus;
+    private netConnected = false;
+
+    constructor() {
+        this.previousAppState = AppState.currentState;
+    }
+
+    public init = async (serverCredentials: ServerCredential[]) => {
+        this.netConnected = Boolean((await NetInfo.fetch()).isConnected);
+        await Promise.all(
+            serverCredentials.map(
+                ({serverUrl, token}) => async () => {
+                    const {database} = DatabaseManager.serverDatabases[serverUrl];
```

Comment:
you should check that `DatabaseManager.serverDatabases[serverUrl]` is not undefined